generator/wrapper_gen.py

```diff
@@ -273,6 +273,26 @@ def exec_func(*args, **keyArgs):
                 reset_daal_context()
 
                 return res
+        elif 'daal4py.oneapi' in sys.modules:
+            import daal4py.oneapi as d4p_oneapi
+            if d4p_oneapi._get_device_name_sycl_ctxt() == 'gpu' and \
+               d4p_oneapi._get_sycl_ctxt_params()['host_offload_on_fail']:
+                import logging
+                try:
+                    res = func(*args, **keyArgs)
+                    logging.info(f"{func.__qualname__.split('.')[0]} successfully run on gpu")
+                    return res
+                except RuntimeError as e:
+                    logging.info(f"{func.__qualname__.split('.')[0]} failed to run on gpu. Fallback to host")
```

Comment:
Good solution - I like it. But it's worth talking about it offline.

Reply:
Talked offline about this feature